test/parallel/xyk-pallet.API.poolVestedToken.test.ts

```diff
@@ -0,0 +1,338 @@
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+
+    const burnUnlockedToken = await burnLiquidity(
+      testUser1.keyRingPair,
+      MGA_ASSET_ID,
+      createdToken,
+      queryInfoResult
+    );
+    expect(getEventResultFromMangataTx(burnUnlockedToken).state).toEqual(
+      ExtrinsicResult.ExtrinsicSuccess
+    );
+
+    const UserBalanceAfterBurningAmount = await api.query.tokens.accounts(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+    expect(
+      UserBalanceAfterBurningAmount.free.sub(
+        UserBalanceAfterBurningAmount.frozen
+      )
+    ).bnEqual(new BN(0));
+
+    expect(
+      UserBalanceBeforeAmount.frozen.sub(UserBalanceAfterBurningAmount.frozen)
+    ).bnGt(new BN(0));
+  });
+
+  test("xyk-pallet- check that all unlocking vesting tokens can be burned", async () => {
+    const api = getApi();
+    const poolLockedValue = toBN("1", 20);
+    const poolBlockValue = toBN("5", 18);
+
+    const VestingTokenFunction = await createPoolAndVestingToken(
+      true,
+      poolLockedValue,
+      poolBlockValue
+    );
+
+    const vestingStartBlockNumber = new BN(
+      VestingTokenFunction.vestingStartBlockNumber
+    );
+
+    const vestingFinishBlockNumber = new BN(
+      vestingStartBlockNumber.add(
+        poolLockedValue.sub(poolBlockValue).div(poolBlockValue)
+      )
+    );
+
+    await waitNecessaryBlock(vestingFinishBlockNumber);
+
+    const UserBalanceBeforeAmount = await api.query.tokens.accounts(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+
+    expect(
+      UserBalanceBeforeAmount.free.sub(UserBalanceBeforeAmount.frozen)
+    ).bnEqual(new BN(0));
+
+    const unlockSomeVestedToken = await unlockVestedToken(
+      testUser1,
+      liquidityID
+    );
+    expect(getEventResultFromMangataTx(unlockSomeVestedToken).state).toEqual(
+      ExtrinsicResult.ExtrinsicSuccess
+    );
+
+    const UserBalanceAfterUnlockingAmount = await api.query.tokens.accounts(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+    expect(
+      UserBalanceAfterUnlockingAmount.free.sub(
+        UserBalanceAfterUnlockingAmount.frozen
+      )
+    ).bnGt(new BN(0));
+
+    //@ts-ignore
+    const queryInfoResult = await api.rpc.xyk.get_max_instant_burn_amount(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+
+    const burnUnlockedToken = await burnLiquidity(
+      testUser1.keyRingPair,
+      MGA_ASSET_ID,
+      createdToken,
+      queryInfoResult
+    );
+    expect(getEventResultFromMangataTx(burnUnlockedToken).state).toEqual(
+      ExtrinsicResult.ExtrinsicSuccess
+    );
+
+    const UserBalanceAfterBurningAmount = await api.query.tokens.accounts(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+
+    expect(UserBalanceAfterBurningAmount.free).bnEqual(new BN(0));
+
+    expect(
+      UserBalanceBeforeAmount.frozen.sub(UserBalanceAfterBurningAmount.frozen)
+    ).bnGt(new BN(0));
```

Comment:
plz check that afterBurning there are no frozen tokens

Reply:
additionaly also validate that the amount of tokens of the new assetId is "defaultCurrencyValue"  and that the use now has all mangata free

Reply:
We check it in L312, because if the free token's value is zero it means that the frozen token's value is zero too.  I will add checking the amount of free Mangata token